sympy/printing/ccode.py

```diff
@@ -393,38 +560,44 @@ def _print_NegativeInfinity(self, expr):
     def _print_NaN(self, expr):
         return 'NAN'
 
-    def _print_fma(self, expr):  # fused mutiply-add
-        return '{0}fma({1}, {2}, {3})'.format(self._ns, *map(self._print, expr.args))
-
-    def _print_log10(self, expr):  # log10 in C89, but type-generic macro in C99
-        return '{0}log10({1})'.format(self._ns, self._print(expr.args[0]))
-
-    def _print_Sqrt(self, expr):
-        return '{0}sqrt({1})'.format(self._ns, self._print(expr.args[0]))
-
-    def _print_Cbrt(self, expr):
-        return '{0}cbrt({1})'.format(self._ns, self._print(expr.args[0]))
-
-    def _print_hypot(self, expr):
-        return '{0}hypot({1}, {2})'.format(self._ns, *map(self._print, expr.args))
-
-    def _print_expm1(self, expr):
-        return '{0}expm1({1})'.format(self._ns, self._print(expr.args[0]))
-
-    def _print_log1p(self, expr):
-        return '{0}log1p({1})'.format(self._ns, self._print(expr.args[0]))
+    # tgamma was already covered by 'known_functions' dict
 
-    def _print_exp2(self, expr):
-        return '{0}exp2({1})'.format(self._ns, self._print(expr.args[0]))
+    @requires(headers={'math.h'}, libraries={'m'})
+    @_as_macro_if_defined
+    def _print_math_func(self, expr, nest=False):
+        known = self.known_functions[expr.__class__.__name__]
+        if not isinstance(known, str):
+            for cb, name in known:
+                if cb(*expr.args):
```

Comment:
Is this new to this PR? Seems to me that anything that's not just a string should be implemented as a printer method.